mllib/src/main/scala/org/apache/spark/ml/feature/VectorAssembler.scala

```diff
@@ -41,6 +42,17 @@ class VectorAssembler(override val uid: String)
 
   def this() = this(Identifiable.randomUID("vecAssembler"))
 
+  /**
+   * Whether to rewrite vector attribute names.
+   * @group param
+   */
+  final val rewriteAttributeNames: BooleanParam =
```

Comment:
Similar arguments here. It would be nice if we can keep ML attributes under the hood. I think the major problem is we tied feature name (or feature group name) with column name, and it is hard to keep good column names during transformation. If in `OneHotEncoder`, we don't add group name to feature name. The attribute transformation is

~~~
county -> OneHotEncoder -> country. : ["US", "CA", ...] -> VectorAssembler -> ["country.US", "country.CA"]
~~~